src/datasources/notebook/datasource.ts

```diff
@@ -76,7 +76,7 @@ export class DataSource extends DataSourceApi<NotebookQuery, NotebookDataSourceO
           throw new Error('The output for the notebook does not match the expected SystemLink format.');
         }
       } else {
-        throw new Error('The notebook failed to execute.');
+        throw new Error(`The notebook failed to execute. Status of the execution: ${execution.status}. Error code: ${execution.errorCode}. Exception: ${execution.exception}.`);
```

Comment:
Do line breaks work here to make the message a bit nicer?

Reply:
Unfortunately this is html, and we escape string like these. So either "\n" nor "<br>" works for this. Tried at least to make the message a bit nicer to read.